raster_array/src/raster_array/raster_array.py

```diff
@@ -328,6 +328,33 @@ def _from_datasetreader(
 
 
 # private helpers --------------------------------------------------------------
+def ensure_band_index(band_index: int | list[int] | None) -> list[int] | None:
+    """Validate and coerce band index(es) to a list of integers.
+
+    Args:
+        band_index (int | list[int] | None): The band index(es) to validate and coerce.
+
+    Returns:
+        list[int] | None: A list of band indices or None if the input is None.
+
+    Raises:
+        TypeError: If the band_index is not an integer, list of integers, or None.
+    """
+    if band_index is None:
+        return band_index
+
+    if isinstance(band_index, int):
+        band_index = [band_index]
```

Comment:
Why check it again on lines 349-353? 
```suggestion
        return [band_index]
```